climt/_components/surface_ice.py

```diff
@@ -212,29 +180,29 @@ def __call__(self, state, timestep):
 
                     # Create temperature profile from cubic spline
                     vertical_coordinates = np.linspace(0, total_height, num_layers)
-                    temp_profile = input_arrays['snow_ice_temperature_poly'][lon, lat](vertical_coordinates)
+                    temp_profile = input_arrays['snow_and_ice_temperature_spline'][lon, lat](vertical_coordinates)
```

Comment:
lon and lat should be renamed to i_lon and i_lat, since they are not actual longitudes and latitudes. This confused me when I first read this part of the code.

Also, you will want to make sure all relevant quantities have "match_dimensions_like" "array_type", so that the number of indices is always accurate.

Reply:
First part handled. Second part will happen when I move to the properties API, maybe in the release following this one.